src/resolve/resolved-manager.c

```diff
@@ -1929,3 +1930,112 @@ void dns_manager_reset_statistics(Manager *m) {
         m->n_failure_responses_served_stale_total = 0;
         zero(m->n_dnssec_verdict);
 }
+
+static int dns_configuration_json(
+                const char *ifname,
+                int ifindex,
+                DnsServer *current_dns_server,
+                DnsServer *dns_servers,
+                DnsSearchDomain *search_domains,
+                sd_json_variant **ret) {
```

Comment:
Not a matter, but by taking the array rather than 'ret', you can use `sd_json_variant_appendbo()`.